lib/active_merchant/billing/gateways/card_connect.rb

```diff
@@ -0,0 +1,283 @@
+        '51' => STANDARD_ERROR_CODE[:card_declined],
+        'C2' => STANDARD_ERROR_CODE[:incorrect_cvc],
+        '54' => STANDARD_ERROR_CODE[:expired_card],
+        '05' => STANDARD_ERROR_CODE[:card_declined],
+        '03' => STANDARD_ERROR_CODE[:config_error],
+        '60' => STANDARD_ERROR_CODE[:pickup_card]
+      }
+
+      def initialize(options = {})
+        requires!(options, :merchant_id, :username, :password)
+
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        if options[:po_number]
+          MultiResponse.run(:use_first_response) do |r|
+            r.process { authorize(money, payment, options) }
+            r.process { capture(money, r.authorization, options) }
+          end
+        else
+          post = {}
+          add_invoice(post, options)
+          add_money(post, money)
+          add_payment(post, payment)
+          add_currency(post, money, options)
+          add_address(post, options)
+          add_customer_data(post, options)
+          add_3DS(post, options)
+          post[:capture] = 'Y'
+          commit('auth', post)
+        end
+      end
+
+      def authorize(money, payment, options = {})
+        post = {}
+        add_money(post, money)
+        add_currency(post, money, options)
+        add_invoice(post, options)
+        add_payment(post, payment)
+        add_address(post, options)
+        add_customer_data(post, options)
+        add_3DS(post, options)
+        commit('auth', post)
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        add_money(post, money)
+        add_reference(post, authorization)
+        add_additional_data(post, options)
+        commit('capture', post)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_money(post, money)
+        add_reference(post, authorization)
+        commit('refund', post)
+      end
+
+      def void(authorization, options = {})
+        post = {}
+        add_reference(post, authorization)
+        commit('void', post)
+      end
+
+      def verify(credit_card, options = {})
+        authorize(0, credit_card, options)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+          .gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]')
+          .gsub(%r(("cvv2\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r(("merchid\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r((&?"account\\":\\")\d*(&?)), '\1[FILTERED]')
+          .gsub(%r((&?"token\\":\\")\d*(&?)), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email] if options[:email]
+      end
+
+      def add_address(post, options)
+        address = options[:billing_address] || options[:address]
+        if address
+          post[:address] = address[:address1] if address[:address1]
```

Comment:
Does it make sense to concatenate `address[:address2]` in any way, or do we not care here?

Reply:
Guess it doesn't hurt to concatenate the two together into the one field. 👍